### ISSUE TITLE
Build for both node and browser

### ISSUE THREAD

**Issue by dvkndn** [dvkndn/tinhte-api-nextjs#1]
Currently the build target is configured to be current node (https://github.com/dvkndn/tinhte-api-nextjs/blob/master/.babelrc#L5)

I have a feeling that this is not correct :( What is the right way here? Since this library is intended to be used in both node (when call api in getInitialProps) and browser (when call api in componentDidMount) env (right?), so:

- Should we have 2 configs (and thus, 2 compiled scripts?), one to be consumed in node, and one for the browser?
- A way to build that supports both?

Current source file: https://github.com/dvkndn/tinhte-api-nextjs/blob/master/src/index.js
Current compiled file: https://github.com/dvkndn/tinhte-api-nextjs/blob/master/lib/index.js (build for node at the moment)

**Comment by dvkndn** [comment]
@daohoangson please help =((

**Comment by daohoangson** [comment]
I think it's fine, Webpack will transpile for browser later in the build pipeline. You probably don't need babel for this library anyway if you are only exporting es6 module though.

While you are at it, probably a good idea to have some tests + simple demo app. 

Btw, I would not commit `lib` into the repo.

**Comment by dvkndn** [comment]
@daohoangson I also think that we don't need `babel` to run in node env.. but maybe in browser env.

I do have a question ragarding "Webpack will transpile for browser later in the build pipeline": since this library is in `node_modules`, I don't think its code will be transpiled.

**Comment by dvkndn** [comment]
Regarding the `lib` I did add it to the gitignore (https://github.com/dvkndn/tinhte-api-nextjs/blob/master/.gitignore#L2) but it seems it was commited before that.. let me try remove it then

**Comment by daohoangson** [comment]
You are right about next.js excluding node_modules from transpiling. Just checked my nwb builds, it uses something like these:

```js
{
  "presets": [
    "env", {
      "loose": true,
      "modules": false
    }
  ]
}
```

**Comment by dvkndn** [comment]
Hm unfortunatelly this did not work either. With the above config, the transpiled code still has `regeneratorruntime`, which is not available in node env but only in browser.

Currently I tried again with target as `2 latest chrome version` but it feels too dangerous...

**Comment by daohoangson** [comment]
I think only babel excludes `node_modules`, webpack still processes those when it build the browser bundle so that's good to go.

Anyway, I'm updating the tinhte-api lib for OTT supports and maybe I will just put the cache logic over there.

**Comment by dvkndn** [comment]
1. Yes it's true that only babel excludes, while webpack does not. However I think that means the `babel-loader` in webpack excludes `node_modules`, which still means the code will not be transpiled  😢 

I did try to use the source file (so no transpile in the library), and the result is that the code is bundled but still ES6. It is almost the same as how I wrote it.
---> Therefore, there are things like rest syntax (...) that older browsers don't understand
---> Moreover, there are things like JSX syntax (<Component />) that no browsers understand at all

2. I don't get this :-ss why is this related to caching logic?
